pennylane/templates/subroutines.py

```diff
@@ -17,26 +17,21 @@
 repeatedly.
 """
 #pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
-from collections.abc import Sequence
-
 from pennylane.ops import Beamsplitter, Rotation
-from pennylane.qnode import QuantumFunctionError
-from pennylane.variable import Variable
+from pennylane.templates.utils import (_check_shapes,
+                                       _check_no_variable,
+                                       _check_wires,
+                                       _check_hyperp_is_in_options)
 
 
 def Interferometer(theta, phi, varphi, wires, mesh='rectangular', beamsplitter='pennylane'):
     r"""General linear interferometer, an array of beamsplitters and phase shifters.
 
     For :math:`M` wires, the general interferometer is specified by
     providing :math:`M(M-1)/2` transmittivity angles :math:`\theta` and the same number of
-    phase angles :math:`\phi`, as well as either :math:`M-1` or :math:`M` additional rotation
+    phase angles :math:`\phi`, as well as :math:`M-1` additional rotation
     parameters :math:`\varphi`.
 
-    For the parametrization of a universal interferometer
-    :math:`M-1` such rotation parameters are sufficient. If :math:`M` rotation
-    parameters are given, the interferometer is over-parametrized, but the resulting
-    circuit is more symmetric, which can be advantageous.
```

Comment:
It was a complication since for everything relying on the interferometer, like the cvqnn, we have to describe that option (else a user may be confused why it works). That adds a lot of clutter - I found it easier to just define the interferometer canonically.

Reply:
:+1: